### ISSUE TITLE
LLVM backend SIMD mode doesn't emit i16x8.shr_s for bit shifts

### ISSUE THREAD

**Issue by brion** [emscripten-core/emscripten#8220]
Testing with LLVM upstream backend, as of LLVM commit a77638d5c1fe7833118083d1b and emscripten incoming commit 1a992bbe648b3dbccf7b15f.

I noticed in the disassembly of some of the functions I'm trying to SIMD-optimize for video codecs that rightward bit shifts on i16x8 vectors aren't being emitted using the i16x8.shr_s instruction, but instead are extracting each lane, shifting it with 32-bit arithmetic, and replacing the lanes back.

Test case repo with minimal source, makefile, and build artifacts: https://github.com/brion/emscripten-wasm-bug-shr

```c
__attribute__((noinline))
static int16x8 do_shift(int16x8 px_v)
{
    return (px_v - 1) >> 4;
}
```

```wat
 (func $do_shift (; 10 ;) (type $5) (param $0 v128) (result v128)
  (i16x8.replace_lane 7
   (i16x8.replace_lane 6
    (i16x8.replace_lane 5
     (i16x8.replace_lane 4
      (i16x8.replace_lane 3
       (i16x8.replace_lane 2
        (i16x8.replace_lane 1
         (i16x8.splat
          (i32.shr_s
           (i16x8.extract_lane_s 0
            (local.tee $0
             (i16x8.add
              (local.get $0)
              (i16x8.splat
               (i32.const -1)
              )
             )
            )
           )
           (i32.const 4)
          )
         )
         (i32.shr_s
          (i16x8.extract_lane_s 1
           (local.get $0)
          )
          (i32.const 4)
         )
        )
        (i32.shr_s
         (i16x8.extract_lane_s 2
          (local.get $0)
         )
         (i32.const 4)
        )
       )
       (i32.shr_s
        (i16x8.extract_lane_s 3
         (local.get $0)
        )
        (i32.const 4)
       )
      )
      (i32.shr_s
       (i16x8.extract_lane_s 4
        (local.get $0)
       )
       (i32.const 4)
      )
     )
     (i32.shr_s
      (i16x8.extract_lane_s 5
       (local.get $0)
      )
      (i32.const 4)
     )
    )
    (i32.shr_s
     (i16x8.extract_lane_s 6
      (local.get $0)
     )
     (i32.const 4)
    )
   )
   (i32.shr_s
    (i16x8.extract_lane_s 7
     (local.get $0)
    )
    (i32.const 4)
   )
  )
 )
```

**Comment by tlively** [comment]
Just pushed a fix to LLVM! It turns out the shifts were being unrolled as a "temporary" workaround for a v8 bug that has since been fixed.

**Comment by brion** [comment]
Confirmed I can now build with the shifts, but at runtime it's failing to compile in Chrome Canary.

```
CompileError: WebAssembly.compile(): Compiling wasm function "dav1d_cdef_filter_8x8_wasm" failed: invalid shift amount @+87647
```

Live test with the standalone test compile at https://brionv.com/misc/shr/test.html

**Comment by tlively** [comment]
Yikes, I’ll open this back up until we can figure that out.

**Comment by tlively** [comment]
Ah, I remembered late last night that V8 still had this bug after all, so it was jumping the gun to remove the workaround in LLVM. I added the shift unrolling back to LLVM and created a [V8 bug](https://bugs.chromium.org/p/v8/issues/detail?id=8934) so progress can be tracked (and so I don't forget about it again).

I'll close this issue since it isn't really an emscripten problem, and we don't want too many duplicates floating around on different bug trackers.
